### DISCUSSION TITLE
getBoundingBox(0) giving weird results

### DISCUSSION THREAD

**Discussion by moving-bits** [mapsforge/vtm#1086] [General]
To my understanding (but I may be wrong), getBoundingBox(0) returns the minimum box of the map's current viewport in lat/lon format.

This sometimes works, at other times I get seemingly weird results like a small area around 0/0 or some other coordinates unrelated to the viewport currently being displayed on the map.

Observation is with Mapsforge offline maps, with either a single offline map or in "combined" mode. (Haven't tried online maps yet.)

What is the correct way to get the actual coordinates of top/left and bottom/right corner for the map?

**Comment by moving-bits** [comment]
Example: I used `mMap.animator().animateTo(bounds)` to set the viewport to [N 53° 32.754' · E 013° 14.274',N 53° 34.860' · E 013° 17.567'].

A second later (to allow for animation) I checked the value of `getBoundingBox()`, and it returned S 00° 01.749' · W 000° 01.011',N 00° 01.749' · E 000° 01.011' (although displaying a viewport related to the one I requested, though not including it fully, but that's a different issue).

I a different test with the same target bounds I got this: N 20° 05.481' · E 004° 15.925',N 20° 08.371' · E 004° 17.704' - both is completely unrelated to the bounds I requested the map to zoom to.

**Comment by devemux86** [comment]
`Map.getBoundingBox(0)` should be the method,
although we haven't seen anything strange so far.

**Comment by devemux86** [comment]
Are you sure there isn't a thread issue?

**Reply by moving-bits** [reply]
this is the code part from the fragment:

```
public void zoomToBounds(final BoundingBox bounds) {
  if (bounds.getLatitudeSpan() == 0 && bounds.getLongitudeSpan() == 0) {
      mMap.animator().animateTo(new GeoPoint(bounds.getMaxLatitude(), bounds.getMaxLongitude()));
  } else {
      if (mMap.getWidth() == 0 || mMap.getHeight() == 0) {
          //See Bug #14948: w/o map width/height the bounds can't be calculated
          // -> postphone animation to later on UI thread (where map width/height will be set)
          mMap.post(() -> mMap.animator().animateTo(bounds));
      } else {
          mMap.animator().animateTo(bounds);
      }
  }
  ActivityMixin.postDelayed(() -> {
      BoundingBox bb = getBoundingBox();
      Log.e("bounds: " + GeopointFormatter.format(GeopointFormatter.Format.LAT_LON_DECMINUTE, Geopoint.forE6(bb.minLatitudeE6, bb.minLongitudeE6))
          + "," + GeopointFormatter.format(GeopointFormatter.Format.LAT_LON_DECMINUTE, Geopoint.forE6(bb.maxLatitudeE6, bb.maxLongitudeE6)));
  }, 1000);
}

(the last block only does some formatting to the results of `getBoundingBox()`)

**Comment by devemux86** [comment]
Can you create a small code example showing the report?
Then we can test it with VTM samples.

If it is reproducible, it would be easier to check what is going on.

When using `setMapPosition` instead of animation, does it work better?

**Reply by moving-bits** [reply]
I've changed the code to use `setMapPosition()`, but that does not seem to have any different result:

```
public void zoomToBounds(final BoundingBox bounds) {
    logBounds("requested: ", bounds);
    if (bounds.getLatitudeSpan() == 0 && bounds.getLongitudeSpan() == 0) {
        mMap.animator().animateTo(new GeoPoint(bounds.getMaxLatitude(), bounds.getMaxLongitude()));
    } else {
        if (mMap.getWidth() == 0 || mMap.getHeight() == 0) {
            //See Bug #14948: w/o map width/height the bounds can't be calculated
            // -> postphone animation to later on UI thread (where map width/height will be set)
            mMap.post(() -> {
                // mMap.animator().animateTo(bounds);
                MapPosition mp = mMap.getMapPosition();
                mp.setByBoundingBox(bounds, mMap.getWidth(), mMap.getHeight());
                mMap.setMapPosition(mp);
            });
        } else {
            // mMap.animator().animateTo(bounds);
            MapPosition mp = mMap.getMapPosition();
            mp.setByBoundingBox(bounds, mMap.getWidth(), mMap.getHeight());
            mMap.setMapPosition(mp);
        }
    }
    ActivityMixin.postDelayed(() -> logBounds("result: ", getBoundingBox()), 1000);
}

private void logBounds(final String info, final BoundingBox bb) {
    Log.e(info + GeopointFormatter.format(GeopointFormatter.Format.LAT_LON_DECMINUTE, Geopoint.forE6(bb.minLatitudeE6, bb.minLongitudeE6))
            + "," + GeopointFormatter.format(GeopointFormatter.Format.LAT_LON_DECMINUTE, Geopoint.forE6(bb.maxLatitudeE6, bb.maxLongitudeE6)));
}
```

Logcat output is:
```
[main] requested: N 53° 32.894' · E 013° 15.233',N 53° 33.818' · E 013° 16.501'
[main] result: S 00° 01.749' · W 000° 01.011',N 00° 01.749' · E 000° 01.011'
```

(but I do get useful results sometimes, so it may be a timing issue. Will retest with a longer delay for the second logging call)

**Reply by devemux86** [reply]
Maybe a conversion issue between degrees vs microdegrees?
